apps/scratch/src/hooks/useEmbeddedScratch.ts

```diff
@@ -105,11 +107,18 @@ export const useEmbeddedScratch = (
                 vm.runtime.once(
                   "ASSERTIONS_CHECKED",
                   (passedAssertions, failedAssertions) =>
-                    resolve({ passedAssertions, failedAssertions }),
+                    resolve({
+                      passedAssertions,
+                      failedAssertions,
+                    }),
                 );
 
                 // once the project is backed up, run the project
                 vm.greenFlag();
+
+                setTimeout(() => {
+                  vm.stopAll();
+                }, maximumExecutionTimeInMs);
```

Comment:
Here we should check if the VM was still running. If so, we might want to pop up a message to the user saying something along the lines of "we stopped the run, it was taking too long..."